examples/gno.land/p/nt/simpledao/membstore.gno

```diff
@@ -0,0 +1,160 @@
+package simpledao
+
+import (
+	"errors"
+
+	"std"
+
+	"gno.land/p/demo/avl"
+	"gno.land/p/gov/dao"
+)
+
+var (
+	errAlreadyMember        = errors.New("address is already a member")
+	errMissingMember        = errors.New("address is not a member")
+	errInvalidAddressUpdate = errors.New("invalid address update")
+)
+
+type MembStoreOption func(*MembStore)
+
+// WithInitialMembers initializes the simpledao member store
+// with an initial member list
+func WithInitialMembers(members []dao.Member) MembStoreOption {
+	return func(store *MembStore) {
+		for _, m := range members {
+			store.members.Set(m.Address.String(), m)
+		}
+	}
+}
+
+// MembStore implements the dao.MembStore abstraction
+type MembStore struct {
+	members *avl.Tree // std.Address -> dao.Member
+
+	totalVotingPower uint64 // cached value for quick lookups
+	majorityPower    uint64 // cached value for quick lookups
+}
+
+// NewMembStore creates a new simpledao member store
+func NewMembStore(opts ...MembStoreOption) *MembStore {
+	m := &MembStore{
+		members:          avl.NewTree(),
+		totalVotingPower: 0,
+		majorityPower:    0,
+	}
+
+	// Apply the options
+	for _, opt := range opts {
+		opt(m)
+	}
+
+	return m
+}
+
+func (m *MembStore) AddMember(member dao.Member) error {
```

Comment:
what about just having SetMemberPower(addr, power), and if 0, remove?

Reply:
We already have an update API, so I've utilized that instead:

[268021d](https://github.com/gnolang/gno/pull/2581/commits/268021dcf8d9d73e3d547ee058b15fa3dfeca2a8)